azkaban-web-server/src/main/java/azkaban/webapp/servlet/LoginAbstractAzkabanServlet.java

```diff
@@ -388,6 +388,11 @@ protected boolean hasPermission(final Project project, final User user,
 
     for (final String roleName : user.getRoles()) {
       final Role role = userManager.getRole(roleName);
+
+      if (role == null) {
+          continue;
```

Comment:
What are the consequences of this? Are users then able to access the resource or do they not have permission to access any resources?

Reply:
You mean the role being null? 
This check just avoids a NullPointerException that will be throwed if code gets to the next if condition, as it tries to call role.getPermission(), so i'm skipping to the next iteration of for loop if role == null
In pratice, the user would not have access to the page, but would receive an HTTP500 response, instead of a treated error

Reply:
Should we throw a 403 Forbidden error instead?

Hmm.. I'm wondering if there's any way to bubble up this error sooner. Like, when the user tries to login, it fails because of an improper role. I'd like to see this fail more quickly instead of when a user tries to access a resource. Maybe we can make a similar request at login time to confirm user role and, if it fails, stop login and throw an error?

Have any thoughts along those lines? What do you think @HappyRay @kunkun-tang ?

Reply:
I agree, this should fail before. I just did this way as `getRole` method was documented "Returns the user role. This may return null.", so i thought that would be nice to make this check.